File: arrays and hashing/Valid Anagram.py
def isAnagram(s: str, t: str) -> bool:
    chars = {}

    for char in s:
        if char in chars:
            chars[char] += 1
        else:
            chars[char] = 1

    for char in t:
        if char in chars:
            chars[char] -= 1
            if chars[char] == -1:
                return False
        else:
            return False

    return True

File: arrays and hashing/Valid Anagram.cs
public class SolutionIsAnagram
{
    public bool IsAnagram(string s, string t)
    {
        if (s.Length != t.Length)
            return false;

        Dictionary<char, int> mapS = new Dictionary<char, int>();
        foreach (char c in s)
        {
            if (mapS.ContainsKey(c))
                mapS[c]++;
            else
                mapS[c] = 1;
        }
        Dictionary<char, int> mapT = new Dictionary<char, int>();
        foreach (char c in t)
        {
            if (mapT.ContainsKey(c))
                mapT[c]++;
            else
                mapT[c] = 1;
        }

        foreach ((char k, int v) in mapS)
        {
            if (mapT.ContainsKey(k))
            {
                if (mapT[k] != v)
                    return false;
            }
            else
                return false;
        }
        return true;
    }
}
